src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -393,7 +393,7 @@ public override IOperation VisitUnaryOperator(IUnaryOperation operation, int? ca
                 return VisitConditionalExpression(operation.Operand, sense: false, captureIdForResult, fallToTrueOpt: null, fallToFalseOpt: null);
             }
 
-            return base.VisitUnaryOperator(operation, captureIdForResult);
+            return new UnaryOperatorExpression(operation.OperatorKind, Visit(operation.Operand, captureIdForResult), operation.IsLifted, operation.IsChecked, operation.OperatorMethod, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
```

Comment:
>captureIdForResult [](start = 96, length = 18)

This is incorrect. The result of the operation is the UnaryOperatorExpression, not its operand. Therefore, we should not be propagating  ```captureIdForResult``` when we are visiting the operand. It sis supposed to be used only for the final result.